### PR TITLE
fix: avoid applying rollout temperature to critic values

### PR DESCRIPTION
**Pull request by Baiyu-Su**
## What

Avoid applying `rollout_temperature` to critic value-head outputs.

`get_responses()` is shared by response-aligned policy-logit extraction and value extraction. Temperature scaling is needed for policy logits when reconstructing rollout log probabilities, but critic values are scalar predictions and should not be divided by the rollout sampling temperature.

## Changes

- Add an `apply_temperature` flag to `get_responses()`.
- Keep temperature scaling enabled by default for existing policy/logprob paths.
- Disable temperature scaling in `get_values()`.
- Add a zero-GPU unit test for non-unit `rollout_temperature`.

## Tested

- `ruff check slime/backends/megatron_utils/loss.py tests/test_value_temperature.py`
- `isort --profile=black --filter-files --check-only slime/backends/megatron_utils/loss.py tests/test_value_temperature.py`
- `black --check slime/backends/megatron_utils/loss.py tests/test_value_temperature.py`
- `PYTHONDONTWRITEBYTECODE=1 python -m py_compile slime/backends/megatron_utils/loss.py tests/test_value_temperature.py`

### CODE REVIEW

**Comment by zhuzilin** [comment]
nice catch!

**Review by zhuzilin — APPROVED** [review]
